backend/de.metas.business/src/main/java/de/metas/inout/impl/InOutQuery.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * #%L
+ * de.metas.business
+ * %%
+ * Copyright (C) 2025 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.inout.impl;
+
+import com.google.common.collect.ImmutableSet;
+import de.metas.document.engine.DocStatus;
+import de.metas.order.OrderId;
+import lombok.Builder;
+import lombok.Singular;
+import lombok.Value;
+
+import javax.annotation.Nullable;
+
+@Value
+@Builder
+public class InOutQuery
```

Comment:
pls also introduce a limit field, e.g.
```java
@Builder.Default private QueryLimit limit = QueryLimit.ONE_HUNDRED;
```

use it in InOutDAO.toSqlQuery.

... just to make sure we are not fetching the whole database by mistake.